modules/developer_manual/pages/general/devenv_docker.adoc

```diff
@@ -0,0 +1,192 @@
+= Setup Your Development Environment based on docker
+
+Using docker containers makes it such easier to create a execution-environment
+for developing. As a developer you nearly don't have to take care about myslq, redis or
+apache configuration. Most of the things are done by container orchestration.
+
+The environment is described in a docker-compose.yml file.
+
+To be able to use and develop ownCloud using docker follow these steps.
+
+1. xref:get-the-source[Get the source]
+2. xref:prerequisite[Prerequisite]
+3.  xref:build-a-docker-container[Build a docker container]
+4.  xref:create-the-environment-with-docker-compose[Create the environment with docker-compose]
+5.  xref:set-user-group-and-permissions[Install software dependencies]
+6.  xref:enable-debug-mode[Enable debug mode]
+7.  xref:setup-owncloud[Setup ownCloud]
+
+[[get-the-source]]
+== Get the source
+Get the owncloud source code from one of the next options:
+
+1. Download a tarball from here https://owncloud.org/download/#owncloud-server-tar-ball
+2. Create a clone of the github repo you can find here https://github.com/owncloud/core
+
+Be aware that cloning the git-repo brings a very raw version of owncloud.
+It must be enriched with some other apps inside the `owncloud/apps`
+folder. It is possible to clone some different apps to that location.
+The `user_management` app for example.
+
+For both cases place the code in a folder inside your home directory
+
+[source,console]
+....
+/home/<your-username>/src/owncloud
+....
+
+for example.
+
+[[prerequisite]]
+== Prerequisite
+
+Let's assume that you already have installed `docker` and `docker-compose` on your computer.
+
+Make sure that you have created a folder parallel to the owncloud sources.
+
+[source,console]
+....
+/home/<your-username>/src/oc_dev_docker
+....
+
+[[build-a-docker-container]]
+=== Build a docker container
```

Comment:
You don't even need that, for CI and development you could use `owncloudci/base` which also includes make and npm.

Reply:
Ok, cool... that make things even easier.

Reply:
Hm... using owncloudi/base leads to following error:

`ERROR: The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing. Continue with the new image? [yN]n ERROR: pull access denied for owncloudi/base, repository does not exist or may require 'docker login'`

using owncloud/base downloads the image but if I try to execute make:

`bash: make: command not found`

Reply:
ok ok dyslexia trap.... owncloud **c** i/base works....

Reply:
I removed the section 'Build a docker container' and modified the image to use in the compose file.